parser/parser.go

```diff
@@ -480,33 +482,33 @@ func (p *parser) VisitExpr(ctx *gen.ExprContext) any {
 // Visit a parse tree produced by CELParser#conditionalOr.
 func (p *parser) VisitConditionalOr(ctx *gen.ConditionalOrContext) any {
 	result := p.Visit(ctx.GetE()).(*exprpb.Expr)
-	b := newBalancer(p.helper, operators.LogicalOr, result)
+	l := p.newLogicManager(operators.LogicalOr, result)
 	rest := ctx.GetE1()
 	for i, op := range ctx.GetOps() {
 		if i >= len(rest) {
 			return p.reportError(ctx, "unexpected character, wanted '||'")
 		}
 		next := p.Visit(rest[i]).(*exprpb.Expr)
 		opID := p.helper.id(op)
```

Comment:
Consider pulling this outside the loop. I think the operator ID only needs to be generated once to avoid creating holes between the arguments' Expr IDs. (i.e: `false || false || false || false` generating 1, 2, 4, 6)

Reply:
In the current format, the IDs will be preserved when the feature is not enabled. As for holes, they'll appear where the operators do, so the non-operator IDs will be consistent.

Reply:
Got it, we can leave this as is.